packages/core-snapshots/src/verifier.ts

```diff
@@ -0,0 +1,72 @@
+import { Models } from "@arkecosystem/core-database";
+import { Blocks, Crypto, Interfaces, Transactions } from "@arkecosystem/crypto";
+
+import * as Exceptions from "./exceptions/verifier";
+
+export class Verifier {
+    public static verifyBlock(block: Models.Block, previousBlock: Models.Block | undefined): void {
+        Verifier.isBlockChained(block, previousBlock);
+        Verifier.verifyBlockSignature(block);
+    }
+
+    public static verifyTransaction(transaction: Models.Transaction): void {
+        Verifier.verifyTransactionSignature(transaction);
+    }
+
+    public static verifyRound(round: Models.Round): void {
+        if (round.publicKey.length !== 66) {
+            throw new Exceptions.RoundVerifyException(round.round.toString());
+        }
+    }
+
+    private static isBlockChained(block: Models.Block, previousBlock: Models.Block | undefined): void {
+        if (!previousBlock) {
+            return;
+        }
+
+        const isBlockChained = block.height - previousBlock.height === 1 && block.previousBlock === previousBlock.id;
+
+        if (!isBlockChained) {
+            throw new Exceptions.BlockNotChainedException(block.id);
+        }
+    }
+
+    private static verifyBlockSignature(blockEntity: Models.Block): void {
+        let isVerified = false;
+
+        try {
+            /* istanbul ignore next */
+            const block = Blocks.BlockFactory.fromData(blockEntity as Interfaces.IBlockData, () => {
+                return blockEntity.timestamp;
+            })!;
+
+            const bytes = Blocks.Serializer.serialize(block.data, false);
+            const hash = Crypto.HashAlgorithms.sha256(bytes);
+
+            isVerified = Crypto.Hash.verifyECDSA(hash, blockEntity.blockSignature, blockEntity.generatorPublicKey);
+        } catch (err) {
+            throw new Exceptions.BlockVerifyException(blockEntity.id, err.message);
+        }
+
+        if (!isVerified) {
+            throw new Exceptions.BlockVerifyException(blockEntity.id, "");
+        }
+    }
+
+    private static verifyTransactionSignature(transaction: Models.Transaction): void {
+        if (transaction.timestamp === 0) {
+            return;
+        }
+
+        let isVerified = false;
+        try {
+            isVerified = Transactions.TransactionFactory.fromBytes(transaction.serialized).isVerified;
+        } catch (err) {
+            throw new Exceptions.TransactionVerifyException(transaction.id, err.message);
+        }
+
+        if (!isVerified) {
+            throw new Exceptions.TransactionVerifyException(transaction.id, "");
```

Comment:
If an argument can be empty it should be marked as optional so it can be omitted rather than having passed in some empty stub value.